src/constants/plugins.js

```diff
@@ -11,3 +11,8 @@ export default {
     PLUGIN_STATE_STOPPING: 5,
     PREPACKAGED_PLUGINS: ['zoom', 'jira', 'mattermost-autolink', 'com.mattermost.nps', 'com.mattermost.custom-attributes', 'github', 'com.mattermost.welcomebot', 'com.mattermost.aws-sns'],
 };
+
+export const MarketplacePluginStatus = {
```

Comment:
@lieut-data @marianunez 

Initially I was thinking we might need more states considering the upgrade case, unfortunately just a single state will not suffice on its own as the client needs more information about the upgrade plugin. It makes sense to switch to using one bit. For upgrade details, I'm thinking a field `Upgrade` can contain details about the new plugin. Do we want admins to have the flexibility to upgrade to any arbitrary version? If so then `Upgrade` has be an array, otherwise a single object will suffice.


Following is my current proposal:
```
{
    "HomepageURL": "https://github.com/mattermost/mattermost-plugin-zoom",
    "DownloadURL": "https://github.com/mattermost/mattermost-plugin-zoom/releases/download/v1.0.7/zoom-1.0.7.tar.gz",
    "SignatureURL": "",
    "Manifest": {
      "id": "zoom",
      "name": "Zoom",
      ...
    },
    "Installed": true,
    Upgrade: {
        "HomepageURL": "https://github.com/mattermost/mattermost-plugin-zoom",
        "DownloadURL": "https://github.com/mattermost/mattermost-plugin-zoom/releases/download/v1.0.7/zoom-1.0.7.tar.gz",
        "SignatureURL": "",
        "Manifest": {
            "id": "zoom",
            "name": "Zoom",
            ...
        },
    }
}
```

Reply:
I'd be inclined to report the current version as a discrete field alongside the marketplace payload. If the current version doesn't match the available version, @marianunez can trigger the `Upgrade` flow and show the proposed text from @asaadmahmood: `<version>, update available: <new version>`.

I think that would give us the ability to infer `installed` as well as upgrade available.

Reply:
I do like having `Version` along side other fields, as long as we name it something like `CurrentVersion` or `InstalledVersion` to separate it from `.Manifest.version`. Could even get rid of `Installed` field if we go down this route 🤔. Thoughts @lieut-data ?


Installed plugin:

```
{
    "HomepageURL": "https://github.com/mattermost/mattermost-plugin-zoom",
    "DownloadURL": "https://github.com/mattermost/mattermost-plugin-zoom/releases/download/v1.0.7/zoom-1.0.7.tar.gz",
    "SignatureURL": "",
    "Manifest": {
        "id": "zoom",
        "name": "Zoom",
        "version": "5.5.5", // NEW
        ...
    },
    "InstalledVersion": "1.2.3",
}
```

NotInstalled Plugin:
```
{
    "HomepageURL": "https://github.com/mattermost/mattermost-plugin-zoom",
    "DownloadURL": "https://github.com/mattermost/mattermost-plugin-zoom/releases/download/v1.0.7/zoom-1.0.7.tar.gz",
    "SignatureURL": "",
    "Manifest": {
        "id": "zoom",
        "name": "Zoom",
        "version": "5.5.5",
        ...
    },
    "InstalledVersion": null, // this can be omitted or be null to denote no installed version.
}
```

Reply:
I like the InstalledVersion to infer installed or not. 

However, we would need to add some logic when checking `Installed` vs `Available` versions to make sure we don't offer upgrade if the version is lower.

Reply:
I'm thinking when `.Manifest.version` is < `.InstalledVersion` then `.Manifest.version` becomes the same as `.InstalledVersion` then that way you  only have to worry about upgrade cases

Reply:
Alright updated pr. https://github.com/mattermost/mattermost-server/pull/11977 is now setting `InstalledVersion` when installed e.g. `"InstalledVersion": "1.2.3"`  and setting an empty string when plugin is not installed e.g. `"InstallVersion": ""`. Went with an `""` empty string since it's not a valid value in our context anyway, if it did then would've made sense to use `null`.